cartography/models/semgrep/findings.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass
+
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeProperties
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class SemgrepSCAFindingNodeProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef('id')
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+    rule_id: PropertyRef = PropertyRef('ruleId', extra_index=True)
+    repository: PropertyRef = PropertyRef('repositoryName', extra_index=True)
+    summary: PropertyRef = PropertyRef('title')
+    description: PropertyRef = PropertyRef('description')
+    package_manager: PropertyRef = PropertyRef('ecosystem')
+    severity: PropertyRef = PropertyRef('severity')
+    cve_id: PropertyRef = PropertyRef('cveId', extra_index=True)
+    reachability_check: PropertyRef = PropertyRef('reachability')
+    reachability_condition: PropertyRef = PropertyRef('reachableIf')
+    reachability: PropertyRef = PropertyRef('exposureType')
+    dependency: PropertyRef = PropertyRef('matchedDependency')
+    dependency_fix: PropertyRef = PropertyRef('closestSafeDependency')
+    ref_urls: PropertyRef = PropertyRef('ref_urls')
+    dependency_file: PropertyRef = PropertyRef('dependencyFileLocation_path')
+    dependency_file_url: PropertyRef = PropertyRef('dependencyFileLocation_url')
+    scan_time: PropertyRef = PropertyRef('openedAt')
+
+
+@dataclass(frozen=True)
+class SemgrepDeploymentProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef('id')
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+    name: PropertyRef = PropertyRef('name')
+    slug: PropertyRef = PropertyRef('slug')
```

Comment:
Add extra indexes

Reply:
I don't think we need to index deployment based on name and Slug, since it will be a single node, but let me know if there is something besides performance that I'm not seeing.